### PR TITLE
GREAT-65: API create sport information

### PR DESCRIPTION
**Pull request by phantuananh18**
Testing result:
2023-01-16_11h50_49

DB changes:
2023-01-16_11h50_22
2023-01-16_11h51_37

### CODE REVIEW

**Review by QuocHuy178 — APPROVED** [review]

File: src/Services/GTT/GTT.Api/SportsManagement/CreateSportsFunction.cs [hunk 1]
```diff
@@ -15,55 +15,57 @@
 
 namespace GTT_API.ExcerciseGroupManagement
 {
-    public class CreateExerciseGroup
+    public class CreateSportsFunction
```
Comment (ClearActionConsistent) [inline src/Services/GTT/GTT.Api/SportsManagement/CreateSportsFunction.cs]: Remove "Function" to make it consistent with other APIs
Create is usually in singular form, not plural

File: src/Services/GTT/shared/GTT.Application/Commands/CreateSports.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
+using FluentValidation;
+using GTT.Application.Requests;
+using GTT.Application.Response;
+using MediatR;
+using System.Net;
+using GTT.Application.Interfaces.Repositories;
+
+namespace GTT.Application.Commands
+{
+    public class CreateSports
```
Comment (ClearActionConsistent) [inline src/Services/GTT/shared/GTT.Application/Commands/CreateSports.cs]: the name should be CreateSport(single form as we create only one Sport)

File: src/Services/GTT/shared/GTT.Application/Interfaces/Repositories/ISportsRepository.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using GTT.Application.Requests;
+
+namespace GTT.Application.Interfaces.Repositories
+{
+    public interface ISportsRepository
+    {
+        Task<int> CreateSports(CreateSportRequestModel request);
```
Comment (ClearActionConsistent) [inline src/Services/GTT/shared/GTT.Application/Interfaces/Repositories/ISportsRepository.cs]: apply singular form for Create Sport

File: src/Services/GTT/shared/GTT.Infrastructure/Repositories/SportsRepository.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            _connection = dbConnectionFactory.CreateConnection();
+        }
+
+        public SportsRepository(IDbConnection connection)
+        {
+            _connection = connection;
+        }
+        #endregion
+
+        public async Task<int> CreateSports(CreateSportRequestModel request)
```
Comment (ClearActionConsistent) [inline src/Services/GTT/shared/GTT.Infrastructure/Repositories/SportsRepository.cs]: - Change function name to CreateSport
- Missing fields: CreatedAt, CreatedBy
- Table should be inclued UpdatedAt and UpdatedBy

File: src/Services/GTT/shared/GTT.Application/Requests/CreateSportRequestModel.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿namespace GTT.Application.Requests
+{
+    public class CreateSportRequestModel
```
Comment (ClearActionConsistent) [inline src/Services/GTT/shared/GTT.Application/Requests/CreateSportRequestModel.cs:3]: Missing field UserId

File: src/Services/GTT/GTT.Api/SportsManagement/CreateSportsFunction.cs (hunk 1 above)
Comment (phantuananh18) [reply src/Services/GTT/GTT.Api/SportsManagement/CreateSportsFunction.cs]: I named this for the purpose of distinguishing between Azure Function and Command. Please help to review @ClearActionConsistent

File: src/Services/GTT/shared/GTT.Application/Requests/CreateSportRequestModel.cs (hunk 5 above)
Comment (phantuananh18) [reply src/Services/GTT/shared/GTT.Application/Requests/CreateSportRequestModel.cs:3]: Where can I get the field UserId from? Can I generate UserId by Guuid support in .NET.
Please help to review my comment 
@ClearActionConsistent

Comment (ClearActionConsistent) [reply src/Services/GTT/shared/GTT.Application/Requests/CreateSportRequestModel.cs:3]: userId is from jwt token, we didn't implement authorization yet, so just skip it for now

**Pull request merged**